core/core-funcs.el

```diff
@@ -36,12 +36,18 @@
   (run-hooks 'text-mode-hook))
 
 (defun spacemacs//get-package-directory (pkg)
-  "Return the directory of PKG. Return nil if not found."
-  (let ((elpa-dir (file-name-as-directory package-user-dir)))
-    (when (file-exists-p elpa-dir)
-      (let* ((pkg-match (concat (symbol-name pkg) "-[0-9]+"))
-             (dir (car (directory-files elpa-dir 'full pkg-match))))
-        (when dir (file-name-as-directory dir))))))
+  "Return the installed directory of PKG. Return `nil' if not found."
+  (let ((directories (cons package-user-dir
```

Comment:
After examining the value of package-alist, it appears that this variable already has the correct paths, i.e., 'dir' field of package-desc struct points to site-lisp directory rather than package-user-dir.

This brings up a question I had for a while.  Why does spacemacs not use package-alist to figure out the location of the installed package, and instead have its own code to search the file system?